File: modules/file-transfer-core/src/main/java/com/shterneregen/securelan/filetransfer/quickshare/QuickShareEvent.java
package com.shterneregen.securelan.filetransfer.quickshare;

import java.util.Objects;

public record QuickShareEvent(
        String shareId,
        QuickShareSnapshot snapshot,
        String message,
        String remoteAddress
) {
    public QuickShareEvent {
        shareId = shareId == null ? "" : shareId;
        Objects.requireNonNull(message, "message must not be null");
        remoteAddress = remoteAddress == null ? "" : remoteAddress;
    }
}

File: modules/file-transfer-core/src/main/kotlin/com/shterneregen/securelan/filetransfer/quickshare/QuickShareEvent.kt
package com.shterneregen.securelan.filetransfer.quickshare

import java.util.Objects

class QuickShareEvent(
    shareId: String?,
    snapshot: QuickShareSnapshot?,
    message: String?,
    remoteAddress: String?,
) {
    private val shareIdValue: String = shareId ?: ""
    private val snapshotValue: QuickShareSnapshot? = snapshot
    private val messageValue: String = Objects.requireNonNull(message, "message must not be null")!!
    private val remoteAddressValue: String = remoteAddress ?: ""

    fun shareId(): String = shareIdValue

    fun snapshot(): QuickShareSnapshot? = snapshotValue

    fun message(): String = messageValue

    fun remoteAddress(): String = remoteAddressValue

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is QuickShareEvent) return false
        return shareIdValue == other.shareIdValue &&
            snapshotValue == other.snapshotValue &&
            messageValue == other.messageValue &&
            remoteAddressValue == other.remoteAddressValue
    }

    override fun hashCode(): Int = Objects.hash(shareIdValue, snapshotValue, messageValue, remoteAddressValue)

    override fun toString(): String =
        "QuickShareEvent[shareId=$shareIdValue, snapshot=$snapshotValue, message=$messageValue, remoteAddress=$remoteAddressValue]"
}
